File: src/main/java/com/lucas/matches/infrastructure/document/SummaryTypeDocument.java
package com.lucas.matches.infrastructure.document;

import java.util.HashMap;
import java.util.Map;

public enum SummaryTypeDocument {
    AVB("AvB"),
    AVBTIME("AvBTime");

    private final String summaryTypeText;
    private static final Map<String, SummaryTypeDocument> map = new HashMap<>();
    SummaryTypeDocument(String summaryTypeText) {
        this.summaryTypeText = summaryTypeText;
    }
    static {
        for (SummaryTypeDocument summaryType : SummaryTypeDocument.values()) {
            map.put(summaryType.summaryTypeText, summaryType);
        }
    }

    public static SummaryTypeDocument value(String summaryTypeText) {
        return map.get(summaryTypeText);
    }
}

File: src/main/kotlin/com/lucas/matches/infrastructure/document/SummaryTypeDocument.kt
package com.lucas.matches.infrastructure.document

enum class SummaryTypeDocument(private val summaryTypeText: String) {
    AVB("AvB"), AVBTIME("AvBTime");

    companion object {
        private val map: MutableMap<String, SummaryTypeDocument> = HashMap()
        fun value(summaryTypeText: String): SummaryTypeDocument? {
            return map[summaryTypeText]
        }

        init {
            for (summaryType in values()) {
                map[summaryType.summaryTypeText] =
                    summaryType
            }
        }
    }
}